addon/modifiers/basic-dropdown-trigger.ts

```diff
@@ -0,0 +1,137 @@
+import Modifier from 'ember-modifier';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+import hasMoved from '../utils/has-moved';
+import { Dropdown } from '../components/basic-dropdown';
+
+interface Args {
+  positional: []
+  named: {
+    dropdown: Dropdown
+    eventType?: 'click' | 'mousedown'
+    stopPropagation?: boolean
+  }
+}
+
+export default class DropdownTriggerModifier extends Modifier<Args> {
+  @tracked private toggleIsBeingHandledByTouchEvents: boolean = false
+  @tracked private touchMoveEvent?: TouchEvent
+
+  didInstall() {
+    if(!this.element.getAttribute('role')) this.element.setAttribute('role', 'button');
+
+    this.element.addEventListener('click', this.handleMouseEvent);
+    this.element.addEventListener('mousedown', this.handleMouseEvent);
+    this.element.addEventListener('keydown', this.handleKeyDown);
+    this.element.addEventListener('touchstart', this.handleTouchStart);
+    this.element.addEventListener('touchend', this.handleTouchEnd);
+  }
+
+  didReceiveArguments() {
+    const { dropdown } = this.args.named
+
+    this.element.setAttribute('data-ebd-id', `${dropdown?.uniqueId}-trigger`);
+    this.element.setAttribute('aria-owns', `ember-basic-dropdown-content-${dropdown?.uniqueId}`);
+    this.element.setAttribute('aria-expanded', dropdown?.isOpen ? 'true' : 'false');
+
+    if (dropdown?.disabled) {
+      this.element.setAttribute('tabindex', '0');
+      this.element.setAttribute('aria-disabled', 'true');
+    } else {
+      this.element.removeAttribute('tabindex');
+      this.element.setAttribute('aria-disabled', 'false');
+    }
+  }
+
+  willRemove() {
+    if (typeof document !== 'undefined') document.removeEventListener('touchmove', this._touchMoveHandler);
+
+    this.element.removeEventListener('click', this.handleMouseEvent);
+    this.element.removeEventListener('mousedown', this.handleMouseEvent);
+    this.element.removeEventListener('keydown', this.handleKeyDown);
+    this.element.removeEventListener('touchstart', this.handleTouchStart);
+    this.element.removeEventListener('touchend', this.handleTouchEnd);
+  }
+
+  @action
+  handleMouseEvent(e: MouseEvent): void {
```

Comment:
The `handleMouseDown` and `handleClick` actions were very similar, so I consolidated them into this action

Reply:
If tests pass, I'm happy with this being one.